app/services/bulk_upload/sales/year2022/row_parser.rb

```diff
@@ -266,9 +266,63 @@ class BulkUpload::Sales::Year2022::RowParser
   validates :field_4, format: { with: /\A\d{2}\z/, message: I18n.t("validations.setup.saledate.year_not_two_digits") }, on: :after_log
 
   validates :field_113, presence: { message: I18n.t("validations.not_answered", question: "ownership type") }, on: :after_log
-  validates :field_57, presence: { message: I18n.t("validations.not_answered", question: "shared ownership type") }, if: :shared_ownership?, on: :after_log
-  validates :field_76, presence: { message: I18n.t("validations.not_answered", question: "shared ownership type") }, if: :discounted_ownership?, on: :after_log
-  validates :field_84, presence: { message: I18n.t("validations.not_answered", question: "shared ownership type") }, if: :outright_sale?, on: :after_log
+
+  validates :field_57,
+            inclusion: {
+              in: [2, 16, 18, 24, 28, 30, 31],
+              category: :setup,
+              question: QUESTIONS[:field_57].downcase,
+            },
+            if: proc { field_57.present? && shared_ownership? },
+            on: :before_log
+
+  validates :field_57,
+            presence: {
+              message: I18n.t("validations.not_answered", question: "shared ownership type"),
+            },
+            if: :shared_ownership?,
+            on: :after_log
+
+  validates :field_76,
+            inclusion: {
+              in: [8, 14, 27, 9, 29, 21, 22],
+              category: :setup,
+              question: QUESTIONS[:field_76].downcase,
+            },
+            if: proc { field_76.present? && discounted_ownership? },
+            on: :before_log
+
+  validates :field_76,
+            presence: {
+              message: I18n.t("validations.not_answered", question: "shared ownership type"),
```

Comment:
Should this be `question: "discounted ownership type"` ?

Reply:
good spot, I've changed this now. there were a couple of others which I have also fixed